### PR TITLE
skip libgdiplus related tests when the Linux platform does not have libgdiplus installed

### LINKED ISSUE #2156: request: skip libgdiplus related tests when the Linux platform does not have libgdiplus installed
Not all flavors of linux come with libgdiplus.

Instead of throwing, can we print a warning and skip the tests instead:

https://github.com/dotnet/performance/blob/b72838ebdd172898a72f2f015ff5384db2c64261/src/benchmarks/micro/libraries/System.Drawing/Perf_Image_Load.cs#L62-L74

Currently, each time I run the benchmark suite, I comment out the libgdiplus related tests to avoid:
```
libgdiplus is missing, you can install it by running 'apt-get install libgdiplus'
Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.TypeInitializationException: The type initializer for 'Gdip' threw an exception.
 ---> System.DllNotFoundException: Unable to load shared library 'libgdiplus' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: liblibgdiplus: cannot open shared object file: No such file or directory
   at System.Drawing.SafeNativeMethods.Gdip.GdiplusStartup(IntPtr& token, StartupInput& input, StartupOutput& output)
   at System.Drawing.SafeNativeMethods.Gdip..cctor()
   --- End of inner exception stack trace ---
   at System.Drawing.SafeNativeMethods.Gdip.GdipCreateBitmapFromScan0(Int32 width, Int32 height, Int32 stride, Int32 format, IntPtr scan0, IntPtr& bitmap)
   at System.Drawing.Tests.Perf_Image_Load.ImageTestData.CreateTestImage(ImageFormat format) in /home/tmds/repos/jenkins-scripts/ci/performance/src/benchmarks/micro/libraries/System.Drawing/Perf_Image_Load.cs:line 99
   at System.Drawing.Tests.Perf_Image_Load.ImageTestData..ctor(ImageFormat format) in /home/tmds/repos/jenkins-scripts/ci/performance/src/benchmarks/micro/libraries/System.Drawing/Perf_Image_Load.cs:line 84
   at System.Drawing.Tests.Perf_Image_Load.CreateTestCases() in /home/tmds/repos/jenkins-scripts/ci/performance/src/benchmarks/micro/libraries/System.Drawing/Perf_Image_Load.cs:line 54
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Drawing.Tests.Perf_Image_Load.ImageFormats() in /home/tmds/repos/jenkins-scripts/ci/performance/src/benchmarks/micro/libraries/System.Drawing/Perf_Image_Load.cs:line 21
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at BenchmarkDotNet.Running.BenchmarkConverter.GetValidValuesForParamsSource(Type parentType, String sourceName)
   at BenchmarkDotNet.Running.BenchmarkConverter.GetArgumentsDefinitions(MethodInfo benchmark, Type target, SummaryStyle summaryStyle)+MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at BenchmarkDotNet.Running.BenchmarkConverter.MethodsToBenchmarksWithFullConfig(Type type, MethodInfo[] benchmarkMethods, IConfig config)
   at BenchmarkDotNet.Running.BenchmarkConverter.TypeToBenchmarks(Type type, IConfig config)
   at BenchmarkDotNet.Running.TypeFilter.<>c__DisplayClass1_0.<Filter>b__0(Type type)
   at System.Linq.Enumerable.SelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.ToArray()
   at BenchmarkDotNet.Running.TypeFilter.Filter(IConfig effectiveConfig, IEnumerable`1 types)
   at BenchmarkDotNet.Running.BenchmarkSwitcher.RunWithDirtyAssemblyResolveHelper(String[] args, IConfig config, Boolean askUserForInput)
   at BenchmarkDotNet.Running.BenchmarkSwitcher.Run(String[] args, IConfig config)
   at MicroBenchmarks.Program.Main(String[] args) in /home/tmds/repos/jenkins-scripts/ci/performance/src/benchmarks/micro/Program.cs:line 42
```

cc @adamsitnik

### PR DESCRIPTION
**Pull request by adamsitnik**
fixes #2156

cc @tmds

### CODE REVIEW

File: src/benchmarks/micro/libraries/System.Drawing/Perf_Image_Load.cs [hunk 1]
```diff
@@ -59,21 +59,23 @@ private static ImageTestData[] CreateTestCases()
...
                 Console.WriteLine("libgdiplus is missing, you can install it by running 'apt-get install libgdiplus'");
                 Console.ResetColor();
 
                 throw;
             }
-            catch (Exception) when (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            catch (Exception) when (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                // skip tests on macOS instead of failing
+                // skip tests on Linux and macOS instead of failing
```
Comment (tmds) [inline src/benchmarks/micro/libraries/System.Drawing/Perf_Image_Load.cs:72]: Do you want to print a warning about the test being skipped?

Comment (adamsitnik) [reply src/benchmarks/micro/libraries/System.Drawing/Perf_Image_Load.cs]: good point, I've changed the logic to always print warning but throw only for Lab runs

**Comment by tmds** [comment]
Thanks, Adam!

**Comment by danmoseley** [comment]
@safern did we decide when to pull System.Drawing support for Unix? (It just says [a future release](https://docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/system-drawing-common-windows-only))

If it's .NET 7, we don't need tests for it.

**Comment by safern** [comment]
> If it's .NET 7, we don't need tests for it.

We didn't decide yet but I think we are trying to aim for .NET 7.

**Pull request merged**